src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestConfigureCallGen.generated.txt

```diff
@@ -1,91 +1,124 @@
+    public static global::Microsoft.Extensions.DependencyInjection.IServiceCollection Configure<T>(this global::Microsoft.Extensions.DependencyInjection.IServiceCollection services, global::Microsoft.Extens
+ions.Configuration.IConfiguration configuration)
     {
-        if (typeof(T) == typeof(Program.MyClass))
+        if (configuration is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(configuration));
+        }
+
+        if (typeof(T) == typeof(global::Program.MyClass))
         {
-            return services.Configure<Program.MyClass>(obj =>
+            return services.Configure<global::Program.MyClass>(obj =>
             {
+                if (!global::Microsoft.Extensions.Configuration.Binder.SourceGeneration.Helpers.HasValue(configuration))
+                {
+                    return default;
+                }
+
                 BindCore(configuration, ref obj);
             });
         }
 
         throw new global::System.NotSupportedException($"Unable to bind to type '{typeof(T)}': 'Generator parser did not detect the type as input'");
     }
+}
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Extensions.Configuration;
+    using System.Collections.Generic;
 
-    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    internal static class Helpers
     {
-        if (obj is null)
+        public static void BindCore(IConfiguration configuration, ref List<Int32> obj)
         {
-            throw new global::System.ArgumentNullException(nameof(obj));
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            Int32 element;
+            foreach (IConfigurationSection section in configuration.GetChildren())
+            {
+                if (section.Value is string stringValue1)
+                {
+                    element = int.Parse(stringValue1);
+                    obj.Add(element);
+                }
+            }
         }
 
-        if (configuration["MyString"] is string stringValue1)
+        public static void BindCore(IConfiguration configuration, ref Dictionary<String, String> obj)
         {
-            obj.MyString = stringValue1;
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            String key;
+            foreach (IConfigurationSection section in configuration.GetChildren())
+            {
+                if (section.Key is string stringValue2)
+                {
+                    key = stringValue2;
+                    String element;
+                    if (section.Value is string stringValue3)
+                    {
+                        element = stringValue3;
+                        obj[key] = element;
```

Comment:
Why do we need the extra `element` variable here?

```suggestion
                    if (section.Value is string stringValue3)
                    {
                        obj[key] = stringValue3;
```

Reply:
I plan to address changes to the logical implementation of the binding code as part of https://github.com/dotnet/runtime/issues/83531 (you gave the same feedback on the initial PR). I'd like to keep this PR scoped to proper formatting if that's fine.